src/batch.jl

```diff
@@ -0,0 +1,210 @@
+struct BatchIntegrand{F,Y,X}
+    # in-place function f!(y, x) that takes an array of x values and outputs an array of results in-place
+    f!::F
+    y::Vector{Y}
+    x::Vector{X}
```

Comment:
Maybe for generality, just `y::Y` and `x::X`, with `{F,Y<:AbstractVector,X<:AbstractVector}`